### PR TITLE
OCPBUGS-42587: Update openstack-provision-etcd-on-local-disk-commands.sh script

### PR DESCRIPTION
**Pull request by MiguelCarpio**
This PR is sent to update `openstack-provision-etcd-on-local-disk-commands.sh` script to follow  the steps to [Deploying on OpenStack with rootVolume and etcd on local disk](https://docs.openshift.com/container-platform/4.17/installing/installing_openstack/deploying-openstack-with-rootVolume-etcd-on-local-disk.html) 

As a result, the etcd will be mounted on the new `/var/lib/etcd` filesystem and the openstack test `[sig-installer][Suite:openshift/openstack] The OpenShift cluster runs with etcd on ephemeral local block device` will pass.

### CODE REVIEW

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.18-e2e-openstack-ccpmso-zone

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.18-e2e-openstack-ccpmso-zone

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.17-e2e-openstack-ccpmso-zone 
periodic-ci-shiftstack-ci-release-4.15-e2e-openstack-ccpmso-zone periodic-ci-shiftstack-ci-release-4.16-e2e-openstack-ccpmso-zone

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.16-e2e-openstack-ccpmso-zone

**Comment by rlobillo** [comment]
Thanks @MiguelCarpio for this fix.

Could you please tell me if you confirmed that 98-var-lib-etcd.yaml is the same than the one documented on below link?

https://docs.openshift.com/container-platform/4.17/installing/installing_openstack/deploying-openstack-with-rootVolume-etcd-on-local-disk.html

Thanks!

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.17-e2e-openstack-ccpmso-zone

**Comment by MiguelCarpio** [comment]
> Thanks @MiguelCarpio for this fix.
> 
> Could you please tell me if you confirmed that 98-var-lib-etcd.yaml is the same than the one documented on below link?
> 
> https://docs.openshift.com/container-platform/4.17/installing/installing_openstack/deploying-openstack-with-rootVolume-etcd-on-local-disk.html
> 
> Thanks!

You're welcome. I'm following this guide https://docs.openshift.com/container-platform/4.17/scalability_and_performance/recommended-performance-scale-practices/recommended-etcd-practices.html#move-etcd-different-disk_recommended-etcd-practices. However, I'm seeing both guides have similar steps and get the same result.

**Comment by MiguelCarpio** [comment]
/assign @EmilienM @stephenfin I have removed the step `Replacing the 98-var-lib-etcd MachineConfig with a modified version that removes the logic for creating and syncing the device and that prevents the nodes from rebooting` which is not mounting `/dev/vdb` on `/var/lib/etcd`. 

After removing it the job passes the test `[sig-installer][Suite:openshift/openstack] The OpenShift cluster runs with etcd on ephemeral local block device`

File: ci-operator/step-registry/openstack/provision/etcd-on-local-disk/openstack-provision-etcd-on-local-disk-commands.sh [hunk 1]
```diff
@@ -150,6 +150,7 @@ spec:
           Where=/var/lib/etcd
           Type=xfs
           Options=defaults,prjquota
+          TimeoutSec=120s
```
Comment (mandre) [inline ci-operator/step-registry/openstack/provision/etcd-on-local-disk/openstack-provision-etcd-on-local-disk-commands.sh:153]: Can we also seize this opportunity to update the ignition version to the one documented in the product docs? The docs for 4.15, 4.16 and 4.17 all uses ignition version 3.4.0. We should be using the same (we're using 3.2.0, see [line 120](https://github.com/openshift/release/pull/58846/files#diff-3a8fcf6c8250a5783267e1e3a02a100ec735ec59119e203701bfff775fd2800aR120)).

**Review by mandre — CHANGES_REQUESTED** [review]
There are a lot of differences between the documented process and what CI does. Take a look at the `98-var-lib-etcd` MachineConfig from docs (left) and the proposed change to CI (right). We should replicate what the docs say:

https://www.diffchecker.com/nDjCLPYh/

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.18-e2e-openstack-ccpmso-zone

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.16-e2e-openstack-ccpmso-zone

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-ci-release-4.18-e2e-openstack-ccpmso-zone

File: ci-operator/step-registry/openstack/provision/etcd-on-local-disk/openstack-provision-etcd-on-local-disk-commands.sh (hunk 1 above)
Comment (MiguelCarpio) [reply ci-operator/step-registry/openstack/provision/etcd-on-local-disk/openstack-provision-etcd-on-local-disk-commands.sh]: Done!

**Comment by MiguelCarpio** [comment]
This PR removed some steps that aren't in [Deploying on OpenStack with rootVolume and etcd on local disk](https://docs.openshift.com/container-platform/4.17/installing/installing_openstack/deploying-openstack-with-rootVolume-etcd-on-local-disk.html). The removed steps are:

- Adding additionalBlockDevices to the control plane machineset
- Waiting up to 5 minutes for the CPMS operator to pick up the edit
- function wait_for_cpms_done
- All control plane machines have the etcd block device validation

Finally, the `openstack-test-openstack` suit has passed the test:
`passed: (12.2s) 2024-11-19T03:15:57 "[sig-installer][Suite:openshift/openstack] The OpenShift cluster runs with etcd on ephemeral local block device"` at the **periodic-ci-shiftstack-ci-release-4.18-e2e-openstack-ccpmso-zone** job without affecting previous jobs.

cc @rlobillo @mandre @EmilienM @stephenfin @mdbooth

**Review by mandre — COMMENTED** [review]
This shows it fixes the [4.18-e2e-openstack-ccpmso-zone](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/58846/rehearse-58846-periodic-ci-shiftstack-ci-release-4.18-e2e-openstack-ccpmso-zone/1858668290004488192) job that was [previously failing](https://prow.ci.openshift.org/job-history/gs/test-platform-results/logs/periodic-ci-shiftstack-ci-release-4.18-e2e-openstack-ccpmso-zone) consistently, since https://github.com/openshift/openstack-test/pull/191 merged.
/lgtm

**Pull request merged**